pages/docs/reference/iterators.md

```diff
@@ -0,0 +1,127 @@
+---
+type: doc
+layout: reference
+category: "Collections"
+title: "Iterators"
+---
+
+# Iterators
+
+For traversing collection elements, the Kotlin standard library supports the commonly used mechanism of _iterators_ – objects that provide access to the elements sequentially without exposing the underlying structure of the collection.
+Iterators are useful when you need to process all the elements of a collection one-by-one, for example, print values or make similar updates to them.
+
+Iterators can be obtained for inheritors of the `Iterable<T>` interface, including `Set` and `List`, by calling the `iterator()` function.
+Once you obtain an iterator, it points to the first element of a collection; calling the `next()` function returns this element and moves the iterator position to the following element if it exists.
+Once the iterator passes through the last element, it can no longer be used for retrieving elements; neither can it be reset to any previous position. To iterate through the collection again, create a new iterator.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = listOf("one", "two", "three", "four")
+    val numbersIterator = numbers.iterator()
+    while (numbersIterator.hasNext()) {
+        println(numbersIterator.next())
+    }
+//sampleEnd
+}
+```
+</div>
+
+Another way to go through an `Iterable` collection is the well-known `for` loop. When using `for` on a collection, you obtain the iterator implicitly. So, the following code is equivalent to the example above:
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = listOf("one", "two", "three", "four")
+    for (item in numbers)
+        println(item)
```

Comment:
indent is lost here on the site, so better to surround loop body with { }

Reply:
added {}